packages/venia-ui/lib/components/CheckoutPage/ShippingMethod/shippingMethod.gql.js

```diff
@@ -36,14 +43,19 @@ export const SET_SHIPPING_METHOD = gql`
                 }
                 ...SelectedShippingMethodCheckoutFragment
                 ...PriceSummaryFragment
-                # Intentionally do not re-fetch available shipping methods because
-                #  a) they are wrong in the mutation response
-                #  b) it is expensive to recalculate.
+                # We include the following fragments to avoid extra requeries
+                # after this mutation completes. This all comes down to not
+                # having ids for shipping address objects. With ids we could
+                # merge results.
+                ...ShippingInformationFragment
+                ...AvailableShippingMethodsCheckoutFragment
```

Comment:
Looking at the previous comments in this fragment, we did not fetch available shipping methods on purpose because they can be faulty or are costly to calculate. Has something changed on the mutation level to make them accurate and fast?

Reply:
I was not able to replicate the "faulty" part of that comment. As for the slowness, sadly it is necessary in order to fix the bug where people get stuck in a state where they cannot continue to the next step. This is primarily because we don't get `id` back from `shipping_addresses` which means we cannot gracefully merge the array entries.

The TLDR is that we were blowing away old data when any query/mutation returned part of `shipping_addresses`. Which means it would re-query anyways. So this just gets it done up front.

I urge you to experiment with this. Remove the lines, see what's happening in the network tab.